pydemolib/nessiedemo/flink.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""NessieDemoFlink handles setting up Flink and Iceberg related objects."""
+
+import os
+import re
+import shutil
+from types import TracebackType
+from typing import Any, Tuple, TypeVar
+
+from pyflink.datastream import StreamExecutionEnvironment
+
+from .demo import _Util, NessieDemo
+
+T = TypeVar("T", bound="NessieDemoFlink")
```

Comment:
I learned, that we can use `self: "NessieDemoFlink"` instead of `self: T` with the decl of `T`. Would prefer to not have the `T` anymore. (I initially copied it from "the internet")

Reply:
fixed